### ISSUE TITLE
Error in valid-jsdoc.js

### ISSUE THREAD

**Issue by readme42** [eslint/eslint#2270]
```
TypeError: Cannot call method 'indexOf' of undefined
    at C:\Users\Cornelius\Documents\GitHub\red\node_modules\eslint\lib\rules\valid-jsdoc.js:107:45
```

Kind regards, 
Cornelius

**Comment by nzakas** [comment]
Thanks file filing an issue. In order to investigate, we will need the following information:

1. What version of ESLint are you using?
2. What source code are you linting? Please give exact code if possible.
3. How do you have this rule configured?

**Comment by lo1tuma** [comment]
4.Do you use the default parser?

**Comment by readme42** [comment]
1. ESLint 10.0.0
2. This obviously incorrect comment should reproduce the exception
```javascript
        /**
         * @param fields [Array]
         */
        function crashIt() {
            return "";
        }
```
3. "valid-jsdoc": 1,
4. yes

**Comment by readme42** [comment]
To give it an update, it still exists in 0.19.0

**Comment by nzakas** [comment]
Yup, the issue is still open so it hasn't been dealt with yet.

**Comment by gyandeeps** [comment]
I think this syntax is incorrect, with that being said It should flag as an syntax error rather than error out. After investigating more I found out the library `doctrine` doesnt thow an syntax error.
I have opened an issue with them asking the question here : https://github.com/Constellation/doctrine/issues/106
If they fix this on their side (ideally they should) then we dont need to make any change, if not then we need to handle this.

**Comment by nzakas** [comment]
It's not a syntax error - that's valid JSDoc (the type is not required). This is a bug in our rule that should be fixed.

**Comment by gyandeeps** [comment]
but its an syntax error if you do the same for `returns` tag.
That's why I thought this might be a bad syntax.

**Comment by nzakas** [comment]
I'm not sure about `returns`, but I know `param` definitely doesn't require a type. We are probably throwing an error because we are assuming a type will be there.

**Comment by gyandeeps** [comment]
what should be expected when this rule is applied on this code (for sure the rule code should not error)

```js
        /**
         * @param fields [Array]
         */
        function crashIt() {
            return "";
        }
```

**Comment by nzakas** [comment]
Ah, yeah, that's definitely a parser bug.

**Comment by andreas-marschke** [comment]
Problem appears when tag.name hasn't been parsed by `doctrine` fix is in the PR #2983

**Comment by gyandeeps** [comment]
@Constellation any feedback here?

**Comment by nzakas** [comment]
Can someone check this with Doctrine 0.7.0?

**Comment by gyandeeps** [comment]
for the code:
```js
        /**
         * @param fields [Array]
         */
        function crashIt() {
            return "";
        }
```

The rule code doesnt crashes and it gives out warning as expected.
